packages/server/README.md

```diff
@@ -1,226 +1,305 @@
+[Hooks](https://openfeature.dev/docs/reference/concepts/hooks) allow for custom logic to be added at well-defined points of the flag evaluation life-cycle
+Look [here](https://openfeature.dev/ecosystem/?instant_search%5BrefinementList%5D%5Btype%5D%5B0%5D=Hook&instant_search%5BrefinementList%5D%5Bcategory%5D%5B0%5D=Server-side&instant_search%5BrefinementList%5D%5Btechnology%5D%5B0%5D=JavaScript) for a complete list of available hooks.
+If the hook you're looking for hasn't been created yet, see the [develop a hook](#develop-a-hook) section to learn how to build it yourself.
 
-```typescript
-import { JsonValue, Provider, ResolutionDetails } from '@openfeature/js-sdk';
+Once you've added a hook as a dependency, it can be registered at the global, client, or flag invocation level.
 
-// implement the provider interface
-class MyProvider implements Provider {
-  readonly metadata = {
-    name: 'My Provider',
-  } as const;
+```ts
+import { OpenFeature } from "@openfeature/js-sdk";
 
-  resolveBooleanEvaluation(flagKey: string, defaultValue: boolean): ResolutionDetails<boolean> {
-    // resolve a boolean flag value
-  }
-
-  resolveStringEvaluation(flagKey: string, defaultValue: string): ResolutionDetails<string> {
-    // resolve a string flag value
-  }
+// add a hook globally, to run on all evaluations
+OpenFeature.addHooks(new ExampleGlobalHook());
 
-  resolveNumberEvaluation(flagKey: string, defaultValue: number): ResolutionDetails<number> {
-    // resolve a numeric flag value
-  }
+// add a hook on this client, to run on all evaluations made by this client
+const client = OpenFeature.getClient();
+client.addHooks(new ExampleClientHook());
 
-  resolveObjectEvaluation<T extends JsonValue>(flagKey: string, defaultValue: T): ResolutionDetails<T> {
-    // resolve an object flag value
-  }
+// add a hook for this evaluation only
+const boolValue = await client.getBooleanValue("bool-flag", false, { hooks: [new ExampleHook()]});
 ```
 
-See [here](https://openfeature.dev/docs/reference/technologies/server/javascript) for a catalog of available providers.
-
-### Hooks:
-
-Hooks are a mechanism that allow for the addition of arbitrary behavior at well-defined points of the flag evaluation life-cycle. Use cases include validation of the resolved flag value, modifying or adding data to the evaluation context, logging, telemetry, and tracking.
+### Logging
 
-```typescript
-import { OpenFeature, Hook, HookContext } from '@openfeature/js-sdk';
-
-// Example hook that logs if an error occurs during flag evaluation
-export class GlobalDebugHook implements Hook {
-  after(hookContext: HookContext, err: Error) {
-    console.log('hook context', hookContext);
-    console.error(err);
-  }
-}
-```
+The JS SDK will log warning and errors to the console by default.
+This can be behavior can be overridden by passing a custom logger either globally or per client.
+A custom logger must implement the [Logger interface](../shared/src/logger/logger.ts).
```

Comment:
You could mention it's compatible with popular frameworks like winston, but I think you say something like that above, already.